Marlin/src/libs/vector_3.cpp

```diff
@@ -75,13 +75,11 @@ void vector_3::normalize() {
   z *= inv_length;
 }
 
-void vector_3::apply_rotation(matrix_3x3 matrix) {
-  const float resultX = x * matrix.matrix[3 * 0 + 0] + y * matrix.matrix[3 * 1 + 0] + z * matrix.matrix[3 * 2 + 0],
-              resultY = x * matrix.matrix[3 * 0 + 1] + y * matrix.matrix[3 * 1 + 1] + z * matrix.matrix[3 * 2 + 1],
-              resultZ = x * matrix.matrix[3 * 0 + 2] + y * matrix.matrix[3 * 1 + 2] + z * matrix.matrix[3 * 2 + 2];
-  x = resultX;
-  y = resultY;
-  z = resultZ;
+void vector_3::apply_rotation(const matrix_3x3 &matrix) {
+  const float _x = x * matrix.matrix[3 * 0 + 0] + y * matrix.matrix[3 * 1 + 0] + z * matrix.matrix[3 * 2 + 0],
+              _y = x * matrix.matrix[3 * 0 + 1] + y * matrix.matrix[3 * 1 + 1] + z * matrix.matrix[3 * 2 + 1],
+              _z = x * matrix.matrix[3 * 0 + 2] + y * matrix.matrix[3 * 1 + 2] + z * matrix.matrix[3 * 2 + 2];
+  x = _x; y = _y; z = _z;
```

Comment:
This had to be reverted, because we want to use the current `x`, `y`, `z` in the calculations, and if this changes `x` before calculating `y`, and `y` before calculating `z` then it will come out very wrong.